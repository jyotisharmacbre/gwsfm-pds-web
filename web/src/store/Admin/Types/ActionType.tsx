export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
