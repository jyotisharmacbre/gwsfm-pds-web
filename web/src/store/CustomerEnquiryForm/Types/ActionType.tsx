export enum ActionType {
  DEFAULT = 'DEFAULT',
  PROJECT_ADD = 'PROJECT_ADD',
  PROJECT_ADD_ERROR = 'PROJECT_ADD_ERROR',
  PROJECT_EDIT_SUCCESS = 'PROJECT_EDIT_SUCCESS',
  GET_ENQUIRY_OVERVIEW_SUCCESS = 'GET_ENQUIRY_OVERVIEW_SUCCESS',
  GET_ENQUIRY_OVERVIEW_ERROR = 'GET_ENQUIRY_OVERVIEW_ERROR',
  GET_PROJECT_DETAIL_SUCCESS = 'GET_PROJECT_DETAIL_SUCCESS',
  GET_PROJECT_DETAIL_ERROR = 'GET_PROJECT_DETAIL_ERROR',
  RESET_PROJECT_DETAIL_STATE = 'RESET_PROJECT_DETAIL_STATE',
  SET_PROJECT_ID_STATE = 'SET_PROJECT_ID_STATE'
}
