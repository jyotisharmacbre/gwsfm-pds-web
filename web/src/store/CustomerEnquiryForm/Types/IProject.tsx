export interface IProject {
  projectName: string;
  companyId: number;
  headOfProject: string;
  projectManager: string;
  scope: string;
  cnNumber: number;
}
