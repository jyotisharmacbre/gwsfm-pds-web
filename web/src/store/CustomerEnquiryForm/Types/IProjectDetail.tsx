export interface IProjectDetail {
  projectId: string;
  name: string;
  contractorId: string;
  companyId: string;
  headOfProject: string;
  projectOwner: string;
  projectManager: string;
  pmHasExperience: boolean;
  scope: string;
  cnNumber: string;
  status: number;
  engagementId?: number;
  countryId: number;
  currencyId: number;
  probabilityOfWinning: String;
  approxValue: string;
  contractTypeId: string;
  cdmNotifiable: boolean;
  firstAssetWorkedOn?: number;
  secondAssetWorkedOn?: number;
  thirdAssetWorkedOn?: number;
  comment: string;
  otherCompanyName: string;
  otherContractName: string;
  businessUnitId: string;
  divisionId: string;
  weightedTCV: number | undefined;
  soldMargin: number | undefined;
  otherEngagementType: string;
  otherFirstAssetWorkedOn: string;
  otherSecondAssetWorkedOn: string;
  otherThirdAssetWorkedOn: string;
}
