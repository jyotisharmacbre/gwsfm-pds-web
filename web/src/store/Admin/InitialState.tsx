import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
