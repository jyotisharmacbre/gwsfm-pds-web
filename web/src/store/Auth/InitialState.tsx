import { IAuth } from "./Types/IAuth";

export const initialState: IAuth= {
    token: null,
    name: null,
    firstName: null,
    email: null
};
