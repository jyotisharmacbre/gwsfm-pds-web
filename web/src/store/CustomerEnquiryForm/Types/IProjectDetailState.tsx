import { IProjectDetail } from './IProjectDetail';
import { IProject } from './IProject';
import EventType from '../../../enums/EventType';
import Notify from '../../../enums/Notify';

export interface IProjectDetailState {
  form: IProjectDetail;
  enquiryOverview: IProject;
  error: string | null;
  loading: boolean;
  notify: Notify;
  event: EventType;
  enquiryOverviewError: string | null;
}
