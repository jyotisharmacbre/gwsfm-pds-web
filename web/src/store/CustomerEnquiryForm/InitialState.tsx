import { IProjectDetailState } from './Types/IProjectDetailState';
import EventType from '../../enums/EventType';
import Notify from '../../enums/Notify';

export const initialState: IProjectDetailState = {
    form: {
        projectId: '',
        name: '',
        contractorId: '',
        companyId: '',
        headOfProject: '',
        projectOwner: '',
        projectManager: '',
        pmHasExperience: false,
        scope: '',
        cnNumber: '',
        status: 1,
        engagementId: undefined,
        countryId: 0,
        currencyId: 0,
        probabilityOfWinning: '',
        approxValue: '',
        contractTypeId: '',
        cdmNotifiable: false,
        firstAssetWorkedOn: undefined,
        secondAssetWorkedOn: undefined,
        thirdAssetWorkedOn: undefined,
        comment: '',
        otherCompanyName: '',
        otherContractName: '',
        divisionId: '',
        otherDivision: '',
        businessUnitId: '',
        otherBusinessUnit: '',
        weightedTCV: undefined,
        soldMargin: undefined,
        otherEngagementType: '',
        otherFirstAssetWorkedOn: '',
        otherSecondAssetWorkedOn: '',
        otherThirdAssetWorkedOn: ''
    },
    enquiryOverview: {
        projectName: '',
        contractorId: '',
        headOfProject: '',
        projectManager: '',
        scope: '',
        cnNumber: '',
        otherContractName: ''
    },
    error: null,
    loading: false,
    notify: Notify.none,
    event: EventType.none,
    enquiryOverviewError: null
};