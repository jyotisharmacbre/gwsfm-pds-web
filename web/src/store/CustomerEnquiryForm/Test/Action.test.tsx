import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as actions from '../../rootActions';
import { initialState } from '../InitialState';
import {ActionType} from '../../Common/Types/ActionType';
import EventType from '../../../enums/EventType';

const mockStore = configureStore([ thunk ]);
let reduxStore;
const setUpStore = (customerEnquiryInitialState) => {
	reduxStore = mockStore({
		project: customerEnquiryInitialState
	});
};
describe('add user redux', () => {
	beforeEach(() => {
		setUpStore(initialState);
		reduxStore.clearActions();
	});
    it('should dispatch the default action when requested projectId data exists in store', () => {
		initialState.form.projectId = 'TEST';
		setUpStore(initialState);
		const expectedActions = [
			{
				type: ActionType.DEFAULT
			}
		];
		reduxStore.dispatch(actions.getProjectDetail('TEST'));
		expect(reduxStore.getActions()).toEqual(expectedActions);
	});

	it('should dispatch the default action when requested project detail add in store', () => {
		initialState.form.projectId = 'TEST';
		setUpStore(initialState);
		reduxStore.dispatch(actions.projectDetailAdd(initialState.form, EventType.save));
		expect(reduxStore.getActions()).toMatchSnapshot();
	});

	it('should dispatch getEnquiryOverview', () => {
		reduxStore.dispatch(actions.getEnquiryOverview('1'));
		expect(reduxStore.getActions()).toMatchSnapshot();
	});

	
	it('should dispatch projectDetailEdit', () => {
		reduxStore.dispatch(actions.projectDetailEdit(initialState.form, EventType.save));
		expect(reduxStore.getActions()).toMatchSnapshot();
	});

	it('should dispatch resetCustomerEnquiryStateDispatch', () => {	
		reduxStore.dispatch(actions.resetCustomerEnquiryState());
		expect(reduxStore.getActions()).toMatchSnapshot();
	});

	it('should dispatch resetProjectDetailStateToInitial', () => {	
		reduxStore.dispatch(actions.resetProjectDetailStateToInitial());
		expect(reduxStore.getActions()).toMatchSnapshot();
	});

	it('should dispatch changeProjectStatus', () => {	
		reduxStore.dispatch(actions.changeProjectStatus(1));
		expect(reduxStore.getActions()).toMatchSnapshot();
	});

	it('should dispatch setProjectId', () => {	
		reduxStore.dispatch(actions.setProjectId('1'));
		expect(reduxStore.getActions()).toMatchSnapshot();
	});
	it('should dispatch resetProjectDetailState', () => {	
		reduxStore.dispatch(actions.resetProjectDetailState());
		expect(reduxStore.getActions()).toMatchSnapshot();
	});
	it('should dispatch resetProjectDetailState', () => {	
		reduxStore.dispatch(actions.resetProjectDetailState());
		expect(reduxStore.getActions()).toMatchSnapshot();
	});
});
