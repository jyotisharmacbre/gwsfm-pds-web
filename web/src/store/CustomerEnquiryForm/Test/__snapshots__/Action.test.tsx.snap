// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add user redux should dispatch changeProjectStatus 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "CHANGE_PROJECT_STATUS_STATE",
  },
]
`;

exports[`add user redux should dispatch getEnquiryOverview 1`] = `Array []`;

exports[`add user redux should dispatch projectDetailEdit 1`] = `Array []`;

exports[`add user redux should dispatch resetCustomerEnquiryStateDispatch 1`] = `
Array [
  Object {
    "type": "RESET_CUSTOMER_ENQUIRY_STATE",
  },
]
`;

exports[`add user redux should dispatch resetProjectDetailState 1`] = `
Array [
  Object {
    "type": "RESET_PROJECT_DETAIL_STATE",
  },
]
`;

exports[`add user redux should dispatch resetProjectDetailState 2`] = `
Array [
  Object {
    "type": "RESET_PROJECT_DETAIL_STATE",
  },
]
`;

exports[`add user redux should dispatch resetProjectDetailStateToInitial 1`] = `
Array [
  Object {
    "type": "RESET_PROJECT_DETAIL_STATE_TO_INITIAL",
  },
]
`;

exports[`add user redux should dispatch setProjectId 1`] = `
Array [
  Object {
    "type": "SET_PROJECT_ID_STATE",
  },
]
`;

exports[`add user redux should dispatch the default action when requested project detail add in store 1`] = `Array []`;
